Sources/MaliciousSiteProtection/API/APIClient.swift

```diff
@@ -67,20 +68,27 @@ public extension MaliciousSiteDetector {
             static let hashPrefix = "hashPrefix"
         }
 
-        public func url(for requestType: APIRequestType) -> URL {
+        public func queryItems(for requestType: APIRequestType) -> QueryItems {
             switch requestType {
             case .hashPrefixSet(let configuration):
-                endpoint.appendingPathComponent(APIPath.hashPrefix).appendingParameters([
-                    QueryParameter.category: configuration.threatKind.rawValue,
-                    QueryParameter.revision: (configuration.revision ?? 0).description,
-                ])
+                return [QueryParameter.category: configuration.threatKind.rawValue,
```

Comment:
addition to my question for `APIRequestV2` change, why do we need to overwrite url components with an extra query items parameter? the old `url(for:requestType)` allowed any url configuration in 1 place, now we need an extra method for it to work. Also it‘s doing url encoding/decoding several times back and forth which seems suboptimal.
I‘d keep an old implementation allowing to combine url items and passed items unless it‘s really needed. Or at least provide some in-code comments why it must be this way and probably add some assertions validating there‘s no URL query items overwritten.

Reply:
In the new Subscription code we don't build the URL before creating the API request, everything happens inside the init, I tried to use the same approach here but I made to many chanhges. As per your previous comment now the APiRequestV2 joins the QueryParams passed and the one already present in the URL, this allowed me to reverte these changes.